backend/src/akvo/flow_services/util.clj

```diff
@@ -70,3 +75,16 @@
                    (http/request final-request))]
     (timbre/debug "response" response)
     response))
+
+(defn get-published-form
+  "Retrieve published form from S3"
+  [app-id form-id]
+  (let [credentials (select-keys (config/find-config app-id) [:access-key :secret-key :s3bucket])
+        object-key (str "surveys/" form-id ".zip")]
+    (try
+      (with-open [in (ZipInputStream. (io/input-stream (:content (s3/get-object credentials (:s3bucket credentials) object-key))))]
+        (let [file (.getNextEntry in)]
+          (when (= (.getName file) (str form-id ".xml"))
```

Comment:
What happens when a `.zip` doesn't have an `.xml`, we'll return `nil`, right? 🤔

Reply:
If we always publish the `.xml` directly, does it make sense to use the `.zip` ?

Reply:
At some point we stopped publishing XML directly and only send the zip + an addition zip noting the version.

Reply:
yes. when there is no `.xml` we return `nil`.